integration-test/src/test/java/org/airsonic/test/StreamIT.java

```diff
@@ -0,0 +1,49 @@
+package org.airsonic.test;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+
+import java.nio.file.Paths;
+import java.nio.file.Path;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class StreamIT {
+    @Test
+    public void testStreamFlacAsMp3() throws Exception {
+        testFileStreaming("dead");
+    }
+
+    @Test
+    public void testStreamM4aAsMp3() throws Exception {
+        testFileStreaming("dance");
+    }
+
+    @Test
+    public void testStreamMp3() throws Exception {
+        testFileStreaming("piano");
+    }
+
+    private void testFileStreaming(String file) throws Exception {
+        Scanner.uploadToDefaultMusicFolder(
+                Paths.get(this.getClass().getResource("/blobs/stream/" + file + "/input").toURI()),
+                "");
+        Scanner.doScan();
+        String mediaFileId = Scanner.getMediaFilesInMusicFolder().parallelStream()
+                .filter(x -> StringUtils.containsIgnoreCase(x.getTitle(), file))
+                .findAny()
+                .map(x -> x.getId())
+                .orElseThrow(() -> new RuntimeException("no media file id matched"));
+
+        byte[] fromServer = Scanner.getMediaFileData(mediaFileId);
+        String expectedBodyResource = String.format("/blobs/stream/" + file + "/responses/1.dat");
+        byte[] expected = IOUtils.toByteArray(StreamIT.class.getResourceAsStream(expectedBodyResource));
+        if (!StringUtils.isBlank(System.getProperty("IntegrationTestsOutputDirectory", ""))) {
+            Path fromServerPath = Paths.get(System.getProperty("IntegrationTestsOutputDirectory", ""), file + ".dat");
+            FileUtils.writeByteArrayToFile(fromServerPath.toFile(), fromServer);
```

Comment:
Any reason you want to write this file to disk? Are you doing something with it afterwards? I think the in-memory assertion below is enough for the test, no?

Reply:
Mostly because I ran into the issue with the updated files (b/c of ffmpeg/codec changes in the Docker image) and needed to get the outputs to disk. I think it's easier to debug if we keep it?

Reply:
okay sure. usually these "debug" kind of things are kept around only if they're uploaded somewhere as debugging artifacts for CI pipelines. The idea is that they're useless if the tests pass, and if they don't, then a dev needs to change code anywhere, so it's not too difficult to just dump it locally.

but if this is your preference, that's fine. it'll just create and recreate some files locally whenever you run tests, not a big issue unless you're worried about the write cycles of your disk.